mono/metadata/native-library.c

```diff
@@ -295,35 +273,50 @@ remove_cached_module (gpointer key, gpointer value, gpointer user_data)
 }
 
 void
-mono_cached_module_cleanup (void)
+mono_global_loader_cache_init (void)
 {
-	// No need for a transition here since the thread is already detached from the runtime
-	mono_global_loader_data_lock ();
+	if (!global_module_map)
+		global_module_map = g_hash_table_new (g_str_hash, g_str_equal);
 
+#ifdef ENABLE_NETCORE
+	if (!native_library_module_map)
+		native_library_module_map = g_hash_table_new (g_direct_hash, g_direct_equal);
+	if (!native_library_module_blacklist)
+		native_library_module_blacklist = g_hash_table_new (g_direct_hash, g_direct_equal);
+	mono_coop_mutex_init (&native_library_module_lock);
+#endif
+}
+
+void
+mono_global_loader_cache_cleanup (void)
+{
 	if (global_module_map != NULL) {
 		g_hash_table_foreach(global_module_map, remove_cached_module, NULL);
 
 		g_hash_table_destroy(global_module_map);
 		global_module_map = NULL;
 	}
 
-	mono_global_loader_data_unlock ();
+	// No need to clean up the native library hash tables since they're netcore-only, where this is never called
 }
 
 static gboolean
 is_absolute_path (const char *path)
 {
 #ifdef HOST_DARWIN
 	if (!strncmp (path, "@executable_path/", 17) || !strncmp (path, "@loader_path/", 13) ||
-	    !strncmp (path, "@rpath/", 7))
-	    return TRUE;
+		!strncmp (path, "@rpath/", 7))
+		return TRUE;
 #endif
```

Comment:
I realize you only changed whitespace.
But other systems have similar things, like `$ORIGIN`.